### PR TITLE
fix problems with missing classes

### PR DESCRIPTION
**Pull request by JKL98ISR**

### CODE REVIEW

File: deepchecks/tabular/metric_utils/additional_classification_metrics.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def _false_positive_rate_per_class(y_true, y_pred, classes):  # False Positives
     result = []
     for cls in classes:
         y_true_cls, y_pred_cls = np.asarray(y_true) == cls, np.asarray(y_pred) == cls
-        matrix = confusion_matrix(y_true_cls, y_pred_cls)
+        matrix = confusion_matrix(y_true_cls, y_pred_cls, labels=[0, 1])
```
Comment (ItayGabbay) [inline deepchecks/tabular/metric_utils/additional_classification_metrics.py:53]: is [0,1] always the case?

File: deepchecks/tabular/metric_utils/scorers.py [hunk 2]
```diff
@@ -122,7 +122,7 @@
     'tnr': make_scorer(true_negative_rate_metric, averaging_method='binary'),
     'jaccard': make_scorer(jaccard_score, zero_division=0),
     'roc_auc': get_scorer('roc_auc'),
-    'neg_log_loss': get_scorer('neg_log_loss')
+    'neg_log_loss': make_scorer(log_loss, greater_is_better=False, needs_proba=True, labels=[0, 1])
```
Comment (ItayGabbay) [inline deepchecks/tabular/metric_utils/scorers.py:125]: is it still `neg_log_loss` after the `greater_is_better=False`?

File: deepchecks/tabular/metric_utils/additional_classification_metrics.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def _micro_false_positive_rate(y_true, y_pred, classes):
     fp, tn = 0, 0
     for cls in classes:
         y_true_cls, y_pred_cls = np.asarray(y_true) == cls, np.asarray(y_pred) == cls
-        matrix = confusion_matrix(y_true_cls, y_pred_cls)
+        matrix = confusion_matrix(y_true_cls, y_pred_cls, labels=[0, 1])
```
Comment (Nadav-Barak) [inline deepchecks/tabular/metric_utils/additional_classification_metrics.py:62]: I think is always not true..

**Review by Nadav-Barak — CHANGES_REQUESTED** [review]
Lets talk

File: deepchecks/tabular/metric_utils/additional_classification_metrics.py (hunk 1 above)
Comment (Nadav-Barak) [reply deepchecks/tabular/metric_utils/additional_classification_metrics.py:53]: yes

File: deepchecks/tabular/metric_utils/scorers.py (hunk 2 above)
Comment (Nadav-Barak) [reply deepchecks/tabular/metric_utils/scorers.py:125]: + I am not sure if log_loss is the same as neg_log_loss, If not - its bad:
I think can create weird behaviour in some of our checks, we require that scorers also follow the convention that greater is better. even written in our docs

Comment (matanper) [reply deepchecks/tabular/metric_utils/scorers.py:125]: I think we need to wrap the log_loss in lambda:
```
def neg_log_loss(y, y_pred, **kwargs):
  return -log_loss(y, y, y_pred, **kwargs)
```

Comment (JKL98ISR) [reply deepchecks/tabular/metric_utils/scorers.py:125]: this is how sklearn defines neg_log_loss
you can see it by printing get_scorer('neg_log_loss')
I only added the labels

File: deepchecks/tabular/metric_utils/scorers.py [hunk 4]
```diff
@@ -297,20 +298,27 @@ def _run_score(self, model, data: pd.DataFrame, label_col):
...
-                    raise errors.DeepchecksValueError(
-                        f'Scorer returned {len(scores)} scores, but model contains '
-                        f'{len(self.model_classes)} classes. Can\'t proceed')
-                scores = dict(zip(self.model_classes, scores))
-                # Add classes which been seen in the data but are not known to the model
-                scores.update({cls: np.nan for cls in set(self.observed_classes) - set(self.model_classes)})
-            return scores
-        else:
-            return self.scorer(model, data, label_col)
+        # The scores returned are for the observed classes but we want scores of the observed classes
```
Comment (matanper) [inline deepchecks/tabular/metric_utils/scorers.py:311]: Can you fix?
`The scores returned are for the observed classes but we want scores of the observed classes` -> `The scores returned are for the observed classes but we want scores of the model classes`

**Review by matanper — APPROVED** [review]

**Pull request merged**